### DISCUSSION TITLE
VCR session datalogging

### DISCUSSION THREAD

**Discussion by PWPInnovator898** [kriswilk/buzzah#9] [General]
I am beginning to implement and test the datalogging feature I noted a few weeks back.
I will update on my progress as soon as I am able to get initial code functional.
Proposed list of VCR parameters to be logged:

1. DATE/TIME of therapy session
2. ACTUATOR_TYPE
3. ACTUATOR_VOLTAGE
4. ACTUATOR_FREQUENCY
5. AMPLITUDE_MIN
6. AMPLITUDE_MAX
7. TIME_ON
8. TIME_OFF
9. TIME_RELAX
10. JITTER 
11. MIRROR
12. TIME_SESSION
13. Total vibration bursts in session
14. Total cumulative vibration bursts (to be used for LRA lifetime calculations)

Comments welcomed. Thanks again!

**Comment by kriswilk** [comment]
I like the idea of logging session parameters and look forward to seeing how you implement it. How do you plan to keep time? The QT Py board has no concept of "real" date/time without a reference of some kind, especially if it's powered off most of the time. One option is a battery-backup powered real-time clock module. For instance something like this one:

https://www.adafruit.com/product/5188

It could be inserted between the QT Py and the 8-way multiplexer. The code could then read the date/time from it. Of course, you'd have to be able to SET the time, too. Not sure how to do that easily.

Even without setting the exact time, you could use the RTC module as a time reference and store log data in relative terms.

An interesting project, for sure!

**Reply by PWPInnovator898** [reply]
Thank you for the helpful suggestions.  I'll update progress on this once I nail down the h/w to support the data logging features.  My main goal is to minimize added complexity.
